tests/models/test_base.py

```diff
@@ -9,121 +10,199 @@
-                continue
-            if _TORCH_DEVICE == "cuda":
-                torch.cuda.empty_cache()
+    # 256x246
+    rep_shape_list_medium_image = [
+        torch.Size([BATCH_SIZE, 1280, 8, 8]),
+        torch.Size([BATCH_SIZE, 1280, 8, 8]),
+        torch.Size([BATCH_SIZE, 1408, 8, 8]),
+    ]
+    # 384x380
+    rep_shape_list_large_image = [
+        torch.Size([BATCH_SIZE, 1280, 12, 12]),
+        torch.Size([BATCH_SIZE, 1280, 12, 12]),
+        torch.Size([BATCH_SIZE, 1408, 12, 12]),
+    ]
+    shape_list_pre_pool = [
+        rep_shape_list_small_image,
+        rep_shape_list_medium_image,
+        rep_shape_list_large_image,
+    ]
+
+    for idx_backbone, backbone in enumerate(EFFICIENTNET_BACKBONES):
+        if _TORCH_DEVICE == "cuda":
+            torch.cuda.empty_cache()
+        model = BaseFeatureExtractor(backbone=backbone).to(_TORCH_DEVICE)
+        for idx_image in range(len(HEIGHTS)):
             fake_image_batch = torch.rand(
-                size=(BATCH_SIZE, 3, HEIGHTS[ind_image], WIDTHS[ind_image]),
+                size=(BATCH_SIZE, 3, HEIGHTS[idx_image], WIDTHS[idx_image]),
                 device=_TORCH_DEVICE,
             )
+            representations = model(fake_image_batch)
+            assert representations.shape == shape_list_pre_pool[idx_image][idx_backbone]
+            # remove model/data from gpu; then cache can be cleared
+            del fake_image_batch
+            del representations
+        del model
+
+    torch.cuda.empty_cache()  # remove tensors from gpu
+
+
+def test_representation_shapes_vit():
+
+    # loop over different backbone versions and make sure that the resulting
+    # representation shapes make sense
+
+    # 128x128
+    rep_shape_list_small_image = [
+        torch.Size([BATCH_SIZE, 256, 8, 8]),  # vit_b_sam
+    ]
+    # 256x256
+    rep_shape_list_medium_image = [
+        torch.Size([BATCH_SIZE, 256, 16, 16]),
+    ]
+    # 384x384
+    rep_shape_list_large_image = [
+        torch.Size([BATCH_SIZE, 256, 24, 24]),
+    ]
+    shape_list_pre_pool = [
+        rep_shape_list_small_image,
+        rep_shape_list_medium_image,
+        rep_shape_list_large_image,
+    ]
+
+    for idx_backbone, backbone in enumerate(VIT_BACKBONES):
+        for idx_image in range(len(HEIGHTS)):
+            if _TORCH_DEVICE == "cuda":
+                torch.cuda.empty_cache()
             model = BaseFeatureExtractor(
-                backbone=backbone, last_resnet_layer_to_get=-2
+                backbone=backbone, image_size=HEIGHTS[idx_image]
             ).to(_TORCH_DEVICE)
+            fake_image_batch = torch.rand(
+                size=(BATCH_SIZE, 3, HEIGHTS[idx_image], HEIGHTS[idx_image]),
+                device=_TORCH_DEVICE,
+            )
             representations = model(fake_image_batch)
-            assert representations.shape == repres_shape_list_all_but_fc[ind]
+            assert representations.shape == shape_list_pre_pool[idx_image][idx_backbone]
```

Comment:
maybe add a comment? this is like the above but just to say that the rep dim depends on image size?